src/cloudai/workloads/sglang/sglang.py

```
@@ -0,0 +1,202 @@
+    def concurrency(self) -> int:
+        return self.max_concurrency
+
+    @property
+    def tps_per_user(self) -> float | None:
+        if self.concurrency <= 0:
+            return None
+        return self.throughput / self.concurrency
+
+    @model_validator(mode="before")
+    @classmethod
+    def derive_num_prompts(cls, data):
```

Ruff (0.15.5)

[warning] Missing return type annotation for classmethod `derive_num_prompts`

(ANN206)